core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-publish-variant/dot-experiments-publish-variant.component.ts

```diff
@@ -34,42 +36,53 @@ import { DotMessagePipeModule } from '@pipes/dot-message/dot-message-pipe.module
         DotDialogModule,
         FormsModule,
         NgIf,
-        TitleCasePipe
+        TitleCasePipe,
+        AsyncPipe,
+        PercentPipe,
+        TooltipModule
     ],
     standalone: true
 })
-export class DotExperimentsPublishVariantComponent implements OnInit {
-    data: DotResultSimpleVariant[];
-    selectedVariant;
-    dialogActions: DotDialogActions;
-    DEFAULT_VARIANT_ID = DEFAULT_VARIANT_ID;
+export class DotExperimentsPublishVariantComponent {
+    readonly dotMessageService: DotMessageService = inject(DotMessageService);
+    readonly store: DotExperimentsReportsStore = inject(DotExperimentsReportsStore);
 
-    @ViewChild('dialog', { static: true }) dialog: DotDialogComponent;
+    readonly vm$: Observable<VmPromoteVariant> = this.store.promotedDialogVm$.pipe(
+        tap(({ experimentId }) => {
+            this.configDialogActions(experimentId);
+        })
+    );
 
-    @Output() hide = new EventEmitter();
-    @Output() publish = new EventEmitter<string>();
-    constructor(private dotMessageService: DotMessageService) {}
+    selectedVariant: Variant;
 
-    ngOnInit(): void {
-        this.setDialogActions();
-    }
+    dialogActions: DotDialogActions = {};
```

Comment:
Why don't you make this an Observable and use the store?

Reply:
I will leave it as it is. Waiting for the new DotDialog.